### DISCUSSION TITLE
Can I change the default filetype?

### DISCUSSION THREAD

**Discussion by arcticpinecone** [hluk/CopyQ#2157] [Q&A]
Hello! I am really new to 'altering' this app, so bear with me. 
I use Obsidian.md to work my personal knowledge management. I have a synchronized folder with my CopyQ so I can pull up the (.txt) and view it from the app if I want. 

My question is. Does CopyQ have to save everything in the .txt files? Is there any way I can tell it:
Hey, please use copy_0001.md instead of copy_0001.txt
Or I guess now: copyq_20221028122235450.md  

Thank you for this app. I love it and use it every day.

**Comment by dbareis** [comment]
_I'm not answering that question but you haven't specified why you would like this which doesn't help your case._

I would like to ask why the file names are so cryptic in the first place, if I have a tab called "Bungalook", why isn't the file called "tab_Bungalook.dat". If there are some non-filename-safe characters then convert only those to HEX or whatever.  At least have recognisable names...

**Reply by arcticpinecone** [reply]
> _I'm not answering that question but you haven't specified why you would like this which doesn't help your case._

Why are you hijacking my thread for your own agenda?

**Comment by dbareis** [comment]
Why don't you like listening to advice?

**Reply by arcticpinecone** [reply]
I do not understand you at all, to be honest. 

I will try to explain myself better.

When I copy some text somewhere by CopyQ, that clip, goes to an encrypted .dat file. 
I have a separate tab. If something there is something from CopyQ I want to keep, I literally Ctrl-P to paste a copy to this tab.
This tab is pointed to my OneDrive folder for CopyQ-Clips. 

I do not understand how/where to configure CopyQ to store that item as a .md instead of .txt for example. 

Obsidian comes into play as it's just looking at that same OneDrive folder. But it launches .txt files when clicking on them. So I would prefer to see them in .md which would provide multi-use for me. 

So, I wondered if there is some way to 'convert' the .txt file to .md automatically. Or why does CopyQ default to being a .txt file when I do this action? I do not understand how to configure the type of file which CopyQ creates when I paste into that tab in the first place. 

![image](https://user-images.githubusercontent.com/94570993/198859778-63cedf8d-1f0a-4ca0-a96a-90b5dc686935.png)

**Comment by hluk** [comment]
It is not exactly possible. Perhaps the closest solution is:

1. Set CopyQ to show files with `.md` extension as `plain/text` format: 
![image](https://user-images.githubusercontent.com/38054/199294185-655cfb86-d790-49f7-9a51-0459ea7b8730.png)
2. Create a new command which adds menu/toolbar item (or shortcut) to create a new md file/item ([here is how to add the command to CopyQ](http://copyq.readthedocs.io/en/latest/faq.html#how-to-load-shared-commands-and-share-them)):
```ini
[Command]
Command="
    copyq:
    const initialText = '';
    const mapFormat = 'application/x-copyq-itemsync-mime-to-extension-map';

    const fileName = dialog('Enter name');
    if (!fileName) {
        abort();
    }

    var item = {};
    item[mimeText] = initialText;
    item[plugins.itemsync.mimeBaseName] = fileName;
    item[mapFormat] = {};
    item[mapFormat][mimeText] = '.md';
    write(0, item);"
Icon=\xf60f
InMenu=true
Name=New Markdown Item
Shortcut=f1
```
3. Edit the content of the new item.

**Reply by arcticpinecone** [reply]
Thank you so much with your help with this! 🙏 I feel like we are so close to it! 

![image](https://user-images.githubusercontent.com/94570993/199849621-3908b8bc-74cd-4a31-9fe2-7855340caef5.png)
I tried what you suggested, but after entering a name such as `test` creates an empty test.md file in the clipboard top. 

![image](https://user-images.githubusercontent.com/94570993/199849635-284eee4c-960e-4e62-8968-c783d1ea81a8.png)
Here it shows the blank test. Also I highlighted the tab because I was hoping I could send it to that tab too while we're at it. 
I have it named: `&Obsidian`

Having a clip saved as .md to my Obsidian tab + folder sync would be so helpful to my work flow! 💖

**Comment by arcticpinecone** [comment]
So close! Are you still there? @hluk 🙂
